### DISCUSSION TITLE
Import/Migrate entire cluster into a new installation.

### DISCUSSION THREAD

**Discussion by malemieux** [k3s-io/k3s#4953] [Q&A]
We've been using K3s for almost 2 years and we still are new to Kubernetes as whole. 
We have a lot of clusters and most of them are doing well. There is a cluster in particular  on which we are having a lot of issues everytimes we upgrade K3s version. So we would like to simply start a new fresh cluster on the side and import everything from the old cluster to this cluster.  

More info about the cluster : 
The cluster is pretty "vanilla" as a whole. We only added an ingress controller and a cert issuer.
The cluster has about 12 namespaces and and in each one of them there is an app running with configmaps, secrets, etc.

So our question is : Is there any simple way to do this migration in a way that we don't have to reinstall every single apps from manifests ( the manifests are versioned in git repositories) and would it be a recommended practice to do ?

**Comment by ChristianCiach** [comment]
I don't really have an answer to your question, but maybe you could take a look at https://github.com/vmware-tanzu/velero

I have never used it. I just have it in my bookmarks  because the project description sounds potentially useful.
